typescript/sdk/src/router/types.ts

```diff
@@ -97,12 +98,32 @@ export const ForeignDeploymentConfigSchema = z.object({
   foreignDeployment: z.string().optional(),
 });
 
-export const RemoteRouterDomain = z.string();
+export const RemoteRouterDomainOrChainNameSchema = z.string().or(z.number());
+export type RemoteRouterDomainOrChainName = z.infer<
+  typeof RemoteRouterDomainOrChainNameSchema
+>;
```

Comment:
this is a bit verbose
do we. not already have this on the ChainMetadata schema?

Reply:
unfortunately noup, there we just allow wither for an int on the domain Id field
<img width="439" alt="image" src="https://github.com/user-attachments/assets/bced9da8-8b9d-477a-9047-ba477172ab44" />